allow network elements rewrite the version field (#25)
ietf-wg-scone/trone: draft-thoji-scone-trone-protocol.md

This field is encoded as a logarithmically spaced distribution over a range
defined by the TRONE protocol version.

When sent by a QUIC endpoint, the Version field of a TRONE packet is set to
0xTRONE2 and the Rate Signal field is set to 0x3F (63), indicating no rate limit
is in place or that the TRONE protocol is not supported by network elements on
the path. All other values (0x00 through 0x3F for protocol version 0xTRONE1 and
0x00 through 0x3E for protocol version 0xTRONE2) represent the ceiling of rates
being advised by the network element(s) on the path.

For TRONE protocol version 0xTRONE1, the rate limits use a logarithmic scale with:

* Base rate (b_min) = 100 Kbps
* Bitrate at value n = b_min * 10^(n/20)

For TRONE protocol version 0xTRONE2, the rate limits use a logarithmic scale with:

* Bitrate at value n = b_min * 10^((n + 64)/20)

> With two versions combined, bitrates between 100 Kbps and 178 Gbps can be

Reviewer: 178 is at n = 61 
n = 62 should be ~199.53. 

```suggestion
With two versions combined, bitrates between 100 Kbps and 199.5 Gbps can be
```

Author: Thank you!

expressed.

Some notable values in these ranges include:

| Version  | Rate Signal |  Bitrate  |
|:---------|:------------|:----------|
| 0xTRONE1 | 0           | 100 Kbps  |
| 0xTRONE1 | 10          | 316 Kbps  |
| 0xTRONE1 | 20          | 1 Mbps    |
| 0xTRONE1 | 30          | 3.16 Mbps |
| 0xTRONE1 | 40          | 10 Mbps   |
| 0xTRONE1 | 50          | 31.6 Mbps |
| 0xTRONE1 | 60          | 100 Mbps  |
| 0xTRONE2 | 6           | 316 Mbps  |
| 0xTRONE2 | 16          | 1 Gbps    |
| 0xTRONE2 | 26          | 3.16 Gbps |
| 0xTRONE2 | 36          | 10 Gbps   |
| 0xTRONE2 | 46          | 31.6 Gbps |
| 0xTRONE2 | 56          | 100 Gbps  |

> | 0xTRONE2 | 62          | 178 Gbps  |

Reviewer: ```suggestion
| 0xTRONE2 | 62          | 199.5 Gbps  |
```

is_long = packet[0] & 0x80 == 0x80
packet_version = packet[1..5]
if is_long and (packet_version == TRONE1_VERSION or packet_version == TRONE2_VERSION):
  packet_rate_value = packet[0] & 0x3f
  (target_rate_version, target_rate_value) = convert_rate_to_signal(target_rate)
  if target_rate_version < packet_version or target_rate_value < packet_rate_value
    packet[1..5] = target_rate_version
    packet[0] = packet[0] & 0xc0 | target_rate_value
~~~

# Version Interaction {#version-interaction}

The TRONE protocol defines two versions (0xTRONE1 and 0xTRONE2) that cover
different ranges of bitrates. This design allows for:

*  Support for both very low bitrates (down to 100 Kbps) and very high bitrates

>    (up to 178 Gbps)

Reviewer: ```suggestion
   (up to 199.5 Gbps)
```
